### DISCUSSION TITLE
Import Javascript file & using its function in C# code

### DISCUSSION THREAD

**Discussion by nguyenhothanhtam0709** [yantrajs/yantra#62] [General]
Hi Yantra team,

I'm trying to load Javascript file from C# code. But I didn't find any docs about it. 
How can I import and using Javascript function in C# code?

**Comment by ackava** [comment]
Hi, we have listed examples on how to use YantraJS, https://github.com/yantrajs/yantra/wiki/JavaScript-Engine-Example , can you give us example of what you want to achieve?

**Reply by nguyenhothanhtam0709** [reply]
I'm sorry if my explanation is confusing for you. I want to import and using Javascript code in C# code.

For example, I have a file named _script.js_ having a _testFunc_ function like bellow:
```
// function in script.js  
export const testFunc = () => {
      console.log("Hello Yantra!");
 }
```

I want to import and use _testFunc_ in C# code.
How can I import and use this function in C# code like bellow:
```
// C# code
var module = context.ImportModule("./script.js");
module.func();
```

**Reply by ackava** [reply]
```c#
var rootFolder = "... global folder to load node_modules from ... , set this to global npm installed";
var context = new JSModuleContext(rootFolder);

var currentWorkingDirectory = ".... project folder..  that contains local node_modules";
var scriptPath = "... actual script location";
var result = await context.RunAsync(currentWorkingDirectory,  scriptPath);

// result contains exports....
// in your case...

var testFunc = result["testFunc"];

testFunc.InvokeFunction(Arguments.Empty);
```

**Reply by nguyenhothanhtam0709** [reply]
I have followed your example. But I got this error:
**Unhandled exception. YantraJS.Core.JSException: undefined is not a function**

My script.js file exports a function named _testFunc_. 
I'm using package YantraJS.Core v1.2.51 in .Net Core 6.0

**Reply by ackava** [reply]
It seems `export const` wasn't working, I have released new fix that should work now.

**Reply by nguyenhothanhtam0709** [reply]
Thanks you so much. It worked.
